File: src/main/scala/lingoncraft/client/PlayerFinder.scala
package lingoncraft.client

import baritone.api.BaritoneAPI
import com.mojang.logging.LogUtils
import dev.langchain4j.agent.tool.Tool

class PlayerFinder {
  private val log = LogUtils.getLogger

  case class PlayerPosition(x: Double, z: Double)

  @Tool(Array("Get the position of the player"))
  def whereAmI(): PlayerPosition = {
    log.info("whereAmI")
    val player =
      BaritoneAPI.getProvider.getPrimaryBaritone.getPlayerContext.player()
    val x = player.lastRenderX
    val z = player.lastRenderZ
    PlayerPosition(x, z)
  }
}

File: src/main/java/lingoncraft/client/finders/PlayerFinder.java
package lingoncraft.client.finders;

import baritone.api.BaritoneAPI;
import dev.langchain4j.agent.tool.Tool;
import net.minecraft.util.math.BlockPos;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

public class PlayerFinder {
    private static final Logger log = LoggerFactory.getLogger(PlayerFinder.class);

    @Tool({"Get the block position of the player"})
    public BlockPos whereAmI() {
        log.info("whereAmI");
        return new BlockPos(BaritoneAPI.getProvider().getPrimaryBaritone().getPlayerContext().playerFeet());
    }
}